### ISSUE TITLE
CI: Suite-k8s-1.18.K8sServicesTest Checks service across nodes with L7 policy Tests NodePort with L7 Policy: Pod cannot connect to service via tftp (failed in request #/10)

### ISSUE THREAD

**Issue by tgraf** [cilium/cilium#11313]
https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/4658/testReport/Suite-k8s-1/18/K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy/

```
/home/jenkins/workspace/cilium-ginkgo_cilium_master/k8s-1.18-gopath/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:431
Pod "testclient-57gvr" can not connect to service "tftp://192.168.36.11:32377/hello" (failed in request 8/10)
Expected command: kubectl exec -n default testclient-57gvr -- curl --path-as-is -s -D /dev/stderr --fail --connect-timeout 5 --max-time 8 tftp://192.168.36.11:32377/hello -w "time-> DNS: '%{time_namelookup}(%{remote_ip})', Connect: '%{time_connect}',Transfer '%{time_starttransfer}', total '%{time_total}'" 
To succeed, but it failed:
Exitcode: 28 
Stdout:
 	 time-> DNS: '0.000018()', Connect: '0.000033',Transfer '0.000000', total '5.001023'
Stderr:
 	 command terminated with exit code 28
	 

/home/jenkins/workspace/cilium-ginkgo_cilium_master/k8s-1.18-gopath/src/github.com/cilium/cilium/test/k8sT/Services.go:611
```

[18927639_K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy.zip](https://github.com/cilium/cilium/files/4575558/18927639_K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy.zip)

**Comment by jrajahalme** [comment]
This test is currently disabled due to a consistent failure in GKE/Azure/ENI (#11235).

**Comment by jrajahalme** [comment]
Note that there is no proxy redirection for TFTP, so this failure should have nothing to do with L7/proxy redirection.

**Comment by jrajahalme** [comment]
The failing test traffic is seemingly between nodes (testclient-57gvr and 192.168.36.11 are in different nodes), but the host port service load balancing at the destination can pick a backend on either node. It is likely that one of these scenarios (backend in the same node as the source, or backend in the different node than the source) works and the other fails, manifesting as flakiness.

**Comment by joestringer** [comment]
In #11400 we see a similar failure, specifically:

```
Pod "testclient-9stfz" can not connect to service "tftp://[::ffff:192.168.36.11]:30438/hello" (failed in request 5/10)
Expected command: kubectl exec -n default testclient-9stfz -- curl --path-as-is -s -D /dev/stderr --fail --connect-timeout 5 --max-time 8 tftp://[::ffff:192.168.36.11]:30438/hello -w "time-> DNS: '%{time_namelookup}(%{remote_ip})', Connect: '%{time_connect}',Transfer '%{time_starttransfer}', total '%{time_total}'" 
To succeed, but it failed:
Exitcode: 28 
Stdout:
 	 time-> DNS: '0.000016()', Connect: '0.000031',Transfer '0.000000', total '5.001082'
Stderr:
 	 command terminated with exit code 28
	 

/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/k8s-1.11-gopath/src/github.com/cilium/cilium/test/k8sT/Services.go:649
```

(Note that this was for the K8sServicesTest Checks service across nodes Tests NodePort BPF Tests with direct routing Tests NodePort test, not the one reported in the issue description here)

**Comment by jrajahalme** [comment]
Another: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Kernel/1481/testReport/junit/Suite-k8s-1/17/K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy/

**Comment by jrajahalme** [comment]
attached logs: [f9367731_K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy.zip](https://github.com/cilium/cilium/files/4636650/f9367731_K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy.zip)

**Comment by jrajahalme** [comment]
@tgraf Mitigation in #11765 via allowing a single fail of allowed traffic out of each test run of 10 counts on the premise that the problem would be kube proxy node/host port related rather than due to Cilium directly. Rationale of applying this mitigation is that we still need to catch persistent traffic drops and policy regressions with node/host port. Would be nice to get an opinion on if this is appropriate or not.

**Comment by nebril** [comment]
Seems like this is still happening very rarely: 
https://jenkins.cilium.io/job/Ginkgo-CI-Tests-k8s1.18-Pipeline/639/
https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/5149/
https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/5205/

logs from last build:
[42cb51c4_K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy.zip](https://github.com/cilium/cilium/files/4885594/42cb51c4_K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy.zip)

**Comment by jrajahalme** [comment]
Fixed by #12248

**Comment by jrajahalme** [comment]
> Seems like this is still happening very rarely:
> https://jenkins.cilium.io/job/Ginkgo-CI-Tests-k8s1.18-Pipeline/639/
> https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/5149/
> https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/5205/
> 
> logs from last build:
> [42cb51c4_K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy.zip](https://github.com/cilium/cilium/files/4885594/42cb51c4_K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy.zip)
> 
> Can this have similar root cause to #10118 ?

All these runs were started before #12248 was merged.
